src/Our.Umbraco.Vorto/Web/UI/App_Plugins/Vorto/js/vorto.js

```diff
@@ -192,6 +202,10 @@
             }
         });
 
+        $scope.$watch("model.value.values", function () {
```

Comment:
I'd be tempted to do this when the active language changes / when the form is submitted, rather than watching the values collection. Watches can be expensive so it's best to avoid them if possible. Unless there is another reason for needing it?